File: swak-rpc/src/main/java/com/swak/rpc/remote/RemoteHandler.java
package com.swak.rpc.remote;

import java.lang.reflect.InvocationHandler;
import java.lang.reflect.Method;

/**
 * 接口拦截器
 * @author lifeng
 */
public class RemoteHandler implements InvocationHandler {

	@Override
	public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
		System.out.println("invoke service: " + method.getDeclaringClass().getName());
		return null;
	}
}

File: swak-rpc/src/main/java/com/swak/rpc/remote/RemoteHandler.kt
package com.swak.rpc.remote

import com.swak.common.coroutines.taskBlock
import java.lang.reflect.InvocationHandler
import java.lang.reflect.Method

/**
 * 通过 kotlin 来实现
 */
class RemoteHandler : InvocationHandler {

    /**
     * 通过协程来实现
     */
    override fun invoke(proxy: Any?, method: Method?, args: Array<out Any>?): Any? {
        taskBlock(100L) {
            // println("协程")
        }
        return null
    }
}